### PR TITLE
Backlog middleware security bug fix

### PR DESCRIPTION
**Pull request by takashi-uesaka**
I use HttpRequestMessage.Headers.Authorization instead of httpClient.DefaultRequestHeaders.

### CODE REVIEW

File: Owin.Security.Providers/Backlog/BacklogAuthenticationHandler.cs [hunk 1]
```diff
@@ -72,9 +73,12 @@ public BacklogAuthenticationHandler(HttpClient httpClient, ILogger logger)
                 body.Add(new KeyValuePair<string, string>("client_secret", Options.ClientSecret));
 
                 // Get token
-                httpClient.DefaultRequestHeaders.Authorization = null;
-                HttpResponseMessage tokenResponse =
-                    await httpClient.PostAsync(Options.TokenEndpoint, new FormUrlEncodedContent(body));
+                var requestForToken = new HttpRequestMessage(HttpMethod.Post, Options.TokenEndpoint);
```
Comment (kevinchalet) [inline Owin.Security.Providers/Backlog/BacklogAuthenticationHandler.cs:76]: To be consistent with the response variable, you could name it `tokenRequest`.

File: Owin.Security.Providers/Backlog/BacklogAuthenticationHandler.cs [hunk 2]
```diff
@@ -88,9 +92,10 @@ public BacklogAuthenticationHandler(HttpClient httpClient, ILogger logger)
                 string tokenType = (string)response.token_type;
 
                 // Get the Backlog user
-                httpClient.DefaultRequestHeaders.Authorization = new System.Net.Http.Headers.AuthenticationHeaderValue(tokenType, Uri.EscapeDataString(accessToken));
-                HttpResponseMessage graphResponse = await httpClient.GetAsync(
-                    Options.UserInfoEndpoint, Request.CallCancelled);
+                var requestForUser = new HttpRequestMessage(HttpMethod.Get, Options.UserInfoEndpoint);
```
Comment (kevinchalet) [inline Owin.Security.Providers/Backlog/BacklogAuthenticationHandler.cs:95]: `graphRequest`?

**Comment by kevinchalet** [comment]
Make sure to rebase and squash your PR to avoid merging these ugly "merge" commits :smile:

**Comment by jerriep** [comment]
@PinpointTownes Out of curiosity - do you not think it holds value to know who merged the PR? Rebasing does not provide that information. Or am I missing something?

**Comment by kevinchalet** [comment]
By "merge" commits, I meant these ones...

image

... not the "merge pull request" commits that are automatically added when you press the magical green "merge this PR" button :smile: 

image

But yeah, ideally, you'd avoid both ones, as they add unnecessary noise in the git history (specially the first one). Of course, knowing the identity of the person who merged the PR is important, but there's a better way to do that using cherry picking:

image

https://github.com/aspnet/Security/commit/9bb8b6114697f14a42e747acd1d1a41e6eed5af1

Cherry picking allows you to keep the identity of both the author and the committer (basically, you :smile:). It's rather easy to use, as you just have to run `git fetch [fork-address]` + `git cherry-pick [commit-hash]` in your git console to "merge" a remote fork into your local repository ` :smile:

**Comment by jerriep** [comment]
Aah, good info to have, thanks :)

**Comment by takashi-uesaka** [comment]
I fixed valiable names and my ugly remote commit log.So, I close this PR and will open new one.

**Comment by kevinchalet** [comment]
Note that re-opening a PR is usually not needed. You could have removed this commits by simply rebasing :smile:

**Pull request closed without merging**